js/editors/graphEditor.js

```diff
@@ -10,20 +10,42 @@ class GraphEditor {
     this.hovered = null;
     this.dragging = false;
     this.mouse = null;
+  }
+  //function to enable the event listeners to be able to draw the graph 
+  enable()  {
     //Create a private event listener
     this.#addEventListeners();
   }
+  //function to disalbe the event listeners to be able to draw the graph 
+  disable() {
+    this.#removeEventListeners();
+    this.selected = false;
+    this.hovered = false;
+  }
 
   #addEventListeners() {
+    this.boundMouseDown = this.#handleMouseDown.bind(this);
```

Comment:
storing the mouse movement so that I can use the reference for removing the event listeners